### PR TITLE
Keb preview steps

### PR DESCRIPTION
**Pull request by piotrmiskiewicz**
**Description**

Changes proposed in this pull request:
- add steps for preview plan (and kyma 2.0) to KEB provisioning process
- configure reconciler client

What is done:
- [x] JSON schema for provisioning
- [x] steps for provisioning
- [x] e2e provisioning test
- [x] deprovisioning step
- [x] configuration of the reconciler client 
- [x] update process steps
- [x] defined input provider for the preview plan

**Related issue(s)**

### CODE REVIEW

**Comment by piotrmiskiewicz** [comment]
example of payload to KEB to test:
```
{
	"service_id": "47c9dcbf-ff30-448e-ab36-d3bad66ba281",
	"plan_id": "5cb3d976-b85c-42ea-a636-79cadda109a9",
	"context": {
		"globalaccount_id": "e449f875-b5b2-4485-b7c0-98725c0571bf",
		"subaccount_id": "preview23",
			"user_id": "piotr.miskiewicz@sap.com"
	},
	"parameters": {
		"name": "preview23	",
		"kymaVersion": "2.0.0-rc1",
		"overridesVersion": "2.0-rc1"
	}
}```

File: components/kyma-environment-broker/internal/reconciler/client.go [hunk 1]
```diff
@@ -77,24 +88,33 @@ func (c *client) ApplyClusterConfig(cluster Cluster) (*State, error) {
...
 	if err != nil {
 		c.log.Error(err)
 		return err
 	}
 
-	_, err = c.httpClient.Do(request)
+	res, err := c.httpClient.Do(request)
 	if err != nil {
 		c.log.Error(err)
 		return err
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/reconciler/client.go:100]: must be temporary error

File: components/kyma-environment-broker/internal/reconciler/client.go [hunk 2]
```diff
@@ -48,7 +50,7 @@ func (c *client) ApplyClusterConfig(cluster Cluster) (*State, error) {
 
 	reader := bytes.NewReader(reqBody)
 
-	request, err := http.NewRequest("POST", fmt.Sprintf("%s/v1/clusters", c.config.reconcilerURL), reader)
+	request, err := http.NewRequest("POST", fmt.Sprintf("%s/v1/clusters", c.config.URL), reader)
 	if err != nil {
 		c.log.Error(err)
 		return &State{}, err
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/reconciler/client.go:56]: temporary error

File: components/kyma-environment-broker/internal/runtime/provider.go [hunk 3]
```diff
@@ -24,6 +24,8 @@ const (
 	onDemandComponentsURLFormat = "https://storage.googleapis.com/kyma-development-artifacts/%s/kyma-components.yaml"
 )
 
+// todo: a new way of components provider for 2.0
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/runtime/provider.go:27]: remove it

File: components/kyma-environment-broker/internal/reconciler/fake.go [hunk 4]
```diff
@@ -163,8 +158,15 @@ func (c *FakeClient) LastClusterConfig(runtimeID string) (*Cluster, error) {
...
+func (c *FakeClient) IsClusterExists(id string) bool {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	_, exists := c.inventoryClusters[id]
+	return exists
+}
+
 func getLastClusterConfig(cluster *registeredCluster) (*Cluster, error) {
-	clusterConfig, found := cluster.clusterConfigs[int64(len(cluster.clusterConfigs)-1)]
+	clusterConfig, found := cluster.clusterConfigs[int64(1)]
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/reconciler/fake.go:169]: ?

File: docs/kyma-environment-broker/03-03-runtime-operations.md [hunk 5]
```diff
@@ -28,6 +28,9 @@ The provisioning process contains the following steps:
...
 | check_runtime  | Check_Runtime                          | Provisioning             | Checks the status of the Provisioner process and asks the Director for the Dashboard URL if the provisioning in Gardener is finished. |  @piotrmiskiewicz (Team Gopher) | 
+| create_kyma    | Get_Kubeconfig                         | Provisioning             | Gets the kubeconfig (executed only for Kyma 2.0).                                                                                            | @piotrmiskiewicz (Team Gopher) |
+| create_kyma    | Create_Cluster_Configuration           | Reconciler               | Applies the cluster configuration  (executed only for Kyma 2.0).                                       | @piotrmiskiewicz (Team Gopher) |
+| create_kyma    | Check_Cluster_Configuration            | Reconciler               | Checks, if the cluster configuration was applied (executed only for Kyma 2.0).                         | @piotrmiskiewicz (Team Gopher) |
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-03-runtime-operations.md:33]:
```suggestion
| create_kyma    | Check_Cluster_Configuration            | Reconciler               | Checks if the cluster configuration is applied (executed only for Kyma 2.0).                         | @piotrmiskiewicz (Team Gopher) |
```

File: docs/kyma-environment-broker/03-03-runtime-operations.md [hunk 6]
```diff
@@ -28,6 +28,9 @@ The provisioning process contains the following steps:
 | create_runtime | ServiceManagerOverrides                | Service Manager          | Configures overrides with Service Manager credentials.                                                                                          | Team Gopher        |
 | create_runtime | Create_Runtime                         | Provisioning             | Triggers provisioning of a Runtime in the Runtime Provisioner.                                                                                                       | @jasiu001 (Team Gopher)        |
 | check_runtime  | Check_Runtime                          | Provisioning             | Checks the status of the Provisioner process and asks the Director for the Dashboard URL if the provisioning in Gardener is finished. |  @piotrmiskiewicz (Team Gopher) | 
+| create_kyma    | Get_Kubeconfig                         | Provisioning             | Gets the kubeconfig (executed only for Kyma 2.0).                                                                                            | @piotrmiskiewicz (Team Gopher) |
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-03-runtime-operations.md:31]:
```suggestion
| create_kyma    | Get_Kubeconfig                         | Provisioning             | Gets the kubeconfig file (executed only for Kyma 2.0).                                                                                            | @piotrmiskiewicz (Team Gopher) |
```

File: docs/kyma-environment-broker/03-03-runtime-operations.md [hunk 7]
```diff
@@ -52,6 +55,7 @@ The deprovisioning process contains the following steps:
 | De-provision_AVS_Evaluations | AvS            | Done        | Removes external and internal monitoring of Kyma Runtime.                                                  | @jasiu001 (Team Gopher)  |
 | IAS_Deregistration           | Identity Authentication Service | Done | Removes the ServiceProvider from IAS. | @jasiu001 (Team Gopher) |
 | EDP_Deregistration           | Event Data Platform | Done | Removes all entries about SKR from Event Data Platform. | @jasiu001 (Team Gopher) |
+| Deregister_Cluster           | Reconciler     | Done | Removes the cluster from the Reconciler. / @piotrmiskiewicz (Team Gopher) |
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-03-runtime-operations.md:58]:
```suggestion
| Deregister_Cluster           | Reconciler     | Done | Removes the cluster from the Reconciler. | @piotrmiskiewicz (Team Gopher) | 
```

File: docs/kyma-environment-broker/03-03-runtime-operations.md [hunk 8]
```diff
@@ -69,6 +73,8 @@ The upgrade process contains the following steps:
 | Upgrade_Kyma_Initialisation  | Upgrade | Done        | Initializes the `UpgradeOperation` instance with data fetched from the `ProvisioningOperation`. | @ksputo (Team Gopher) |
 | Overrides_From_Secrets_And_Config_Step  | Upgrade | Done        | Builds an input configuration that is passed as overrides to Runtime Provisioner. | @ksputo (Team Gopher) |
 | Upgrade_Runtime              | Upgrade | Done        | Triggers the upgrade of a Runtime in Runtime Provisioner. | @ksputo (Team Gopher) |
+| Get_Kubeconfig               | Upgrade | Done        | Gets a kubeconfig (executed only for Kyma 2.0). | @piotrmiskiewicz (Team Gopher) |
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-03-runtime-operations.md:76]:
```suggestion
| Get_Kubeconfig               | Upgrade | Done        | Gets the kubeconfig file (executed only for Kyma 2.0). | @piotrmiskiewicz (Team Gopher) |
```

File: docs/kyma-environment-broker/08-06-orchestration-status.md [hunk 9]
```diff
@@ -1,4 +1,4 @@
----
+   ---
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/08-06-orchestration-status.md:1]:
```suggestion
---
```

**Review by klaudiagrz — DISMISSED** [review]

**Review by ksputo — APPROVED** [review]

**Review by klaudiagrz — APPROVED** [review]

**Pull request merged**